spec/Generics.tex

```diff
@@ -475,12 +467,84 @@ \section{Generic Types}
 \label{Generic_Types}
 \index{generics!types}
 \index{types!generic}
+
+Generic types comprise built-in generic types, generic classes, and
+generic records.
+
+\subsubsection{Built-in Generic Types}
+\label{Built_in_Generic_types}
+\index{integral (generic type)@\chpl{integral} (generic type)}
+\index{numeric (generic type)@\chpl{numeric} (generic type)}
+\index{enumerated (generic type)@\chpl{enumerated} (generic type)}
+\index{enum (generic type)@\chpl{enum} (generic type)}
+\index{class (generic type)@\chpl{class} (generic type)}
+\index{record (generic type)@\chpl{record} (generic type)}
+
+The types \chpl{integral}, \chpl{numeric} and \chpl{enum}
+are generic types that can only be instantiated with, respectively, the
+signed and unsigned integral types, all of the numeric types, and
+all enumerated types. The type \chpl{enumerated} is currently available
+as a synonym for \chpl{enum}.
+
+The type \chpl{record} can be instantiated with any record type, and
+the type \chpl{class} can be instantiated with any class type.
+
+The memory management strategies described in~\rsec{Class_Types}
+are also generic types that can be instantiated with any class using
+that memory management strategy. These include \chpl{owned} \chpl{shared}
+\chpl{borrowed} and \chpl{unmanaged}.
+
+Note that the \chpl{class} type can combine with \chpl{?}
+(see~\rsec{Nilable_Classes}) and with the management decorators
+(see~\rsec{Class_Types}) to specify a particular variety of class without
+specifying its type:
+
+\begin{itemize}
+\item
+\chpl{class} can instantiate with any non-nilable class using any
+memory management strategy
+
+\item
+\chpl{class?} can instantiate with any class using any memory managemeent
+strategy but will use the nilable variant of that class in an
+instantiation. When used as an argument type, a value of non-nilable class
+type will be implicitly converted to the nilable type on the call. As a
+result, in practice, this type can accept any class type.
```

Comment:
`As a result, a formal of the type \chpl{class?}  can accept an actual of any class type.`